tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-attribute-access-syntax.bal

```diff
@@ -194,6 +194,146 @@ function testErrorsOnXMLIndexedOptionalAttributeAccess() {
     assertError((x3/**/<f>/*)[0]?.id, errorMessage, expTextDetailMessage);
 }
 
+type XC xml:Comment;
+type XPI xml:ProcessingInstruction;
+type XT xml:Text;
+type XE xml:Element;
+
+function testXmlAttributeAccessOnXmlUnionTypes() {
+    xml<xml:Element|xml:Text> x1 = xml `<a attr="aa">a</a>`;
+    assertNonErrorValue(x1.attr, "aa");
+    assertNonErrorValue(x1?.attr, "aa");
```

Comment:
At least for a few of these, we have to declare variables and test. Or we won't be testing typing positive cases.

Reply:
Same for testErrorsOnXmlAttributeAccessOnNonXmlElementValue.